### PR TITLE
CAMEL-14297: Introduce RouteBuilderConfigurer

### PR DESCRIPTION
**Pull request by davsclaus**

### CODE REVIEW

**Comment by davsclaus** [comment]
The test in
org.apache.camel.quarkus.core.CoreTest#testLookupRoutes

has a TODO as the 2nd route from RouteBuilderConfigurer is not discovered.

It uses @Produces annotation from JEE but Camel cannot discover it. Not sure how we can make this possible.
The regular RouteBuilder classes are discovered via jandex index and added during recorder magic.

I would assume a @Produces annotation from CDI/JEE would also work. But since the bean is not injected somewhere then arc may not trigger it. So maybe we need some jandex magic to discover all methods that returns a RouteBuilderConfigurer and are annotated with @Produces should then record the method, or whatever needs to be done.

**Comment by lburgazzoli** [comment]
you can mark the bean as `@Unremovable` to avoid ArC to eliminate it, we can probably also add the `RouteBuilderConfigurer` class to the list of unmoveable beans

**Comment by davsclaus** [comment]
Thanks @lburgazzoli that fixes it

File: integration-tests/core/src/test/java/org/apache/camel/quarkus/core/CoreTest.java [hunk 1]
```diff
@@ -33,12 +33,18 @@
...
+
     @Test
     public void testContainerLookupFromRegistry() {
         RestAssured.when().get("/test/registry/lookup-registry").then().body(is("true"));
         RestAssured.when().get("/test/registry/lookup-context").then().body(is("true"));
     }
 
+    @Test
+    public void testLookupRoutes() {
+        RestAssured.when().get("/test/routes/lookup-routes").then().body(is("true"));
```
Comment (ppalaga) [inline integration-tests/core/src/test/java/org/apache/camel/quarkus/core/CoreTest.java:45]: Comparing the list of routeIds here would make it easier to troubleshoot if something goes wrong.
```suggestion
        RestAssured.when().get("/test/routes/lookup-routes").then().body(is("foo,bar"));
```

Comment (davsclaus) [reply integration-tests/core/src/test/java/org/apache/camel/quarkus/core/CoreTest.java]: Yeah good idea

**Pull request merged**